### ISSUE TITLE
Quotes don't work in terminal inside GVIM

### ISSUE THREAD

**Issue by lucasquin** [vim/vim#10453]
### Steps to reproduce

1. Install gvim on ubuntu with `sudo apt install vim-gtk3`
2. Select the keyboard layout: English(US, alt.intl.) or English(US, alt.intl. with dead keys) 
3. Open Gvim
4. `:term`
5. try typing quotes in the terminal that was opened

### Expected behaviour

Being able to type quotes or any other accent normally.

### Version of Vim

8.2

### Environment

Ubuntu
$TERM =  xterm-256color

**Comment by chrisbra** [comment]
Cannot reproduce and this seems strange. Does it happen when using `:gvim --clean`? Does it happen with a different Desktop Environment or when run by a different user?

**Comment by lucasquin** [comment]
@chrisbra Yes, it still happens using gvim --clean. I'm just using a single user on my Ubuntu. I updated the steps to reproduce the bug, I found that using the keyboard layout as English (US) works normally, but using an English (US, alt. intl.) layout, the quotes don't work, and neither do accents, which is fine strange.

**Comment by gdupras** [comment]
I can reproduce the bug on Kubuntu. It affects gVim but not Vim.

**Comment by lucasquin** [comment]
@gdupras Yes, this only affects gVim. And on windows it works normally.

**Comment by brammool** [comment]
> @gdupras Yes, this only affects gVim. And on windows it works normally.

What GUI is this, GTK 3?

It might be related to input methods.

-- 
Eagles may soar, but weasels don't get sucked into jet engines.

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by gdupras** [comment]
Yes, I'm using GTK 3 too.

>It might be related to input methods.

I think so. Looks like there's an issue with dead keys. With the layout "English (US, alt. intl)", doing "\`" then "e" normally gives "è". This works correctly in a normal buffer in gVim. However, in a terminal window I get "e" instead, without the accent.

With a different layout where you can type "è" directly without a dead key, it works correctly.

**Comment by lucasquin** [comment]
@gdupras exactly.

**Comment by carlosfmesilva** [comment]
I do have exactly the same problem. In my case, I am using a Portuguese keyboard.

```VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Sep 01 2023 19:18:18)
Included patches: 1-1837```

This only happens in vim-gtk3. In regular vim it does not happen.
